clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java

```diff
@@ -78,23 +65,20 @@
   private final URI clusterUri;
   private final String entityIdentifier;
   private final ConcurrentMap<String, ClusteredSpace> knownPersistenceSpaces = new ConcurrentHashMap<>();
-  private final Timeouts timeouts;
-  private final Properties properties;
+  private final ConnectionState connectionState;
 
-  private volatile Connection clusterConnection;
-  private ClusterTierManagerClientEntityFactory entityFactory;
-  private ClusterTierManagerClientEntity entity;
-  private final ConcurrentMap<String, ClusterTierClientEntity> clusterTierEntities = new ConcurrentHashMap<>();
+  private final Set<String> reconnectSet = Collections.newSetFromMap(new ConcurrentHashMap<>());
```

Comment:
It feels like the `reconnectSet` is probably a property of a `ConnectionState` - it's the set of caches that the 'previous' connection had retrieved. In that way the new `ConnectionState` would reconnect using the previous state as a seed.  Not something to fix now but something to consider when looking at cleaning this up.